### PR TITLE
PDA Nuclear Engine checks

### PR DESCRIPTION
**Pull request by TheTyrant25**
[FIX] [FEATURE] [STATION SYSTEMS]
## About the PR
Adds functionality to the PDA powerchecker app. It will now show turbine power output, RPM, stator load, and internal temperature, as well as the reactor radiation level and temperature.
## Why's this needed?
It didn't have it before, and all the other engines have it. Even catalytic generators, lol.
## Changelog
```changelog
(u)Tyrant
(*)You can now check the status of the nuclear reactor and turbines in the PDA powerchecker app!
```

### CODE REVIEW

**Comment by TheTyrant25** [comment]
we're having some small technical difficulties
and i don't know how to check what ~~`MACHINES_POWER` and~~ `MACHINES_FISSION` includes

File: code/obj/item/device/pda2/smallprogs.dm [hunk 1]
```diff
@@ -559,10 +559,12 @@ Code:
 	var/obj/machinery/power/pt_laser/laser
 	var/obj/machinery/power/generatorTemp/generator
 	var/obj/machinery/carouselpower/carousel
+	var/obj/machinery/atmospherics/binary/reactor_turbine/nuke_turbine
+	var/obj/machinery/atmospherics/binary/nuclear_reactor/nuke_reactor
 
 	proc/find_machinery(obj/ref, type)
 		if(!ref || ref.disposed)
-			ref = locate(type) in machine_registry[MACHINES_POWER]
+			ref = locate(type) in machine_registry[MACHINES_POWER || MACHINES_FISSION]
```
Comment (TobleroneSwordfish) [inline code/obj/item/device/pda2/smallprogs.dm:567]: this does not do what you think it does, `3 || 4 == 3`

Comment (frawhst) [reply code/obj/item/device/pda2/smallprogs.dm:567]: `locate(...) in (machine_registry[power] + machine_registry[fission])` should work (pseudocode)

Comment (TheTyrant25) [reply code/obj/item/device/pda2/smallprogs.dm:567]: I see
I have no idea how the machine registry and the defines actually work, see, so i didnt know how to fix it. I'll try this

Comment (TheTyrant25) [reply code/obj/item/device/pda2/smallprogs.dm]: update: this has not worked
Either `locate()` doesn't play nice with the `+`, or the things are not in `MACHINES_FISSION`

Comment (TheTyrant25) [reply code/obj/item/device/pda2/smallprogs.dm]: update: reactors and turbines weren't in `MACHINES_FISSION`. Adding that in fixed the whole thing.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
At least on Nadir, this just doesn't seem to work? Possibly due to the catalytic generators.

**Comment by TheTyrant25** [comment]
> At least on Nadir, this just doesn't seem to work? Possibly due to the catalytic generators.

I reshuffled the order and that seems to have fixed things.

File: code/obj/item/device/pda2/smallprogs.dm [hunk 2]
```diff
@@ -602,27 +608,70 @@ Code:
...
+			for(var/x=1 to length(nuke_reactor.component_grid))
+				if (!isnull(rodlevel))
+					return
+				for(var/y=1 to length(nuke_reactor.component_grid[1]))
+					if(nuke_reactor.component_grid[x][y])
+						if(istype(nuke_reactor.component_grid[x][y],/obj/item/reactor_component/control_rod))
+							var/obj/item/reactor_component/control_rod/CR = nuke_reactor.component_grid[x][y]
+							rodlevel = CR.configured_insertion_level
+							return
+			for some unfathomable reason this broke the whole damn thing, someone else can give it a crack if they like. */
```
Comment (TobleroneSwordfish) [inline code/obj/item/device/pda2/smallprogs.dm:641]: either fix this or remove it, big blocks of commented code shouldn't go in a PR

File: code/obj/item/device/pda2/smallprogs.dm [hunk 3]
```diff
@@ -602,27 +608,70 @@ Code:
...
+			. += "<BR><h4>Reactor Status</h4>"
+			. += "Radiation Level: [engineering_notation(nuke_reactor.radiationLevel)]<BR>"
```
Comment (TobleroneSwordfish) [inline code/obj/item/device/pda2/smallprogs.dm:643]:
```suggestion
			. += "Radiation Level: [engineering_notation(nuke_reactor.radiationLevel)] clicks<BR>"
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/obj/item/device/pda2/smallprogs.dm (hunk 2 above)
Comment (TheTyrant25) [reply code/obj/item/device/pda2/smallprogs.dm:641]: good point, i'll shorten the comment

**Pull request merged**